COMPILING.md

```diff
@@ -0,0 +1,145 @@
+
+- libSDL/liballegro: hardware access (video, sound, mouse)
+- zlib: (de)compressing of old (0.3.0-1.0.5) savegames, content downloads,
+   heightmaps
+- liblzo2: (de)compressing of old (pre 0.3.0) savegames
+- liblzma: (de)compressing of savegames (1.1.0 and later)
+- libpng: making screenshots and loading heightmaps
+- libfreetype: loading generic fonts and rendering them
+- libfontconfig: searching for fonts, resolving font names to actual fonts
+- libicu: handling of right-to-left scripts (e.g. Arabic and Persian) and
+   natural sorting of strings (Linux only)
+
+OpenTTD does not require any of the libraries to be present, but without
+liblzma you cannot open most recent savegames and without zlib you cannot
+open most older savegames or use the content downloading system.
+Without libSDL/liballegro on non-Windows and non-macOS machines you have
+no graphical user interface; you would be building a dedicated server.
+
+## Windows:
+
+You need Microsoft Visual Studio 2015 Update 3 or newer.
+
+You can download the free Visual Studio Community Edition from Microsoft at
+https://visualstudio.microsoft.com/vs/community/.
+
+OpenTTD needs the Platform SDK, if it isn't installed already. This can be
+done during installing Visual Studio, by selecting
+`Visual C++ MFC for x86 and x64` (and possibly
+`Visual C++ ATL for x86 and x64` depending on your version). If not, you
+can get download it as [MS Windows Platform SDK](https://developer.microsoft.com/en-US/windows/downloads/windows-10-sdk).
+
+Install the SDK by following the instructions as given.
+
+Dependencies for OpenTTD on Windows are handled via
+[vcpkg](https://github.com/Microsoft/vcpkg/). First you need to install vcpkg
+by following the `Quick Start` instructions of their
+[README](https://github.com/Microsoft/vcpkg/blob/master/README.md).
+
+After this, you can install the dependencies OpenTTD needs. We advise to use
+the `static` versions, and OpenTTD currently needs the following dependencies:
+
+- liblzma
+- libpng
+- lzo
+- zlib
+
+To install both the x64 (64bit) and x86 (32bit) variants, you can use:
+
+```ps
+.\vcpkg install liblzma:x64-windows-static libpng:x64-windows-static lzo:x64-windows-static zlib:x64-windows-static
+.\vcpkg install liblzma:x86-windows-static libpng:x86-windows-static lzo:x86-windows-static zlib:x86-windows-static
+```
+
+In case you want to build with SDL support
+you need to add WITH_SDL to the project settings.
+
+Open the relevant project file and it should build automatically.
+- VS 2015: projects/openttd_vs140.sln
+- VS 2017: projects/openttd_vs141.sln
+- VS 2019: projects/openttd_vs142.sln
+
+Set the build mode to `Release` in
+`Build > Configuration manager > Active solution configuration`.
+You can now compile.
+
+If everything works well the binary should be in `objs\Win[32|64]\Release\openttd.exe`
+and in `bin\openttd.exe`
+
+You can also build OpenTTD with MSYS/MinGW or Cygwin/MinGW using the Makefile.
```

Comment:
MSYS/MinGW doesn't work, but MSYS2/MinGW-w64 does when following https://wiki.openttd.org/Compiling_on_Windows_using_MSYS2 (hmm we should remove freetype from this wiki page)